src/actions/photosActions.ts

```diff
@@ -33,28 +34,97 @@ const fetchPhotosetRejected = (err: string) => ({
   payload: err,
 });
 
+
 export function downloadPhotos(image_hashes: string[]) {
-  return function () {
-    Server.post(
-      `photos/download`,
-      {
-        image_hashes: image_hashes,
-      },
-      {
-        responseType: "blob",
-      }
-    ).then(response => {
-      const downloadUrl = window.URL.createObjectURL(new Blob([response.data], { type: "application/zip" }));
-      const link = document.createElement("a");
-      link.href = downloadUrl;
-      link.setAttribute("download", "file.zip");
-      document.body.appendChild(link);
-      link.click();
-      link.remove();
+  let fileUrl;
+
+
+  const startDownloadProcess = () => {
+    showNotification({
+      message: "Download Starting ...",
+      title: i18n.t<string>("toasts.downloadstart"),
+      color: "teal",
     });
+    return Server.post("photos/download", {
+      image_hashes: image_hashes,
+    })
+  };
+  const checkDownloadStatus = (job_id) => Server.get(`photos/download?job_id=${job_id}`)
+      .then((response) =>  response.data.status)
+      .catch((error) =>  console.error("Error checking download status:", error));
+ 
+  const getDownloadFile=(fileUrl: string):Promise<AxiosResponse<Blob>>=> axios.get<Blob>(`${serverAddress}/media/zip/${fileUrl}`,{responseType: "blob",})
+
+  const deleteDownloadFile = (fileUrl) => 
+     Server.delete(`delete/zip/${fileUrl}`)
+      .then((response) => response.data.status )
+      .catch((error) => {
+        console.error("Error Deleting the file :", error);
+      });
+ 
+
+  
+  return function () {
+    startDownloadProcess()
+    
+      .then((response) => {
+        fileUrl = response.data.url;
+        let checkStatusInterval;
+
+        const checkStatus = () => {
+         
+          checkDownloadStatus(response.data.job_id)
+            .then((status) => {
+              if (status === "SUCCESS") {
+                clearInterval(checkStatusInterval); // Stop checking once successful
+                  
+                  getDownloadFile(fileUrl).then(response => {
+                    const downloadUrl = window.URL.createObjectURL(new Blob([response.data], { type: "application/zip" }));
+
+                    const link = document.createElement("a");
+                    link.href = downloadUrl;
+                    link.setAttribute("download", "file.zip");
+                    document.body.appendChild(link);
+                    link.click();
+                    link.remove();
+                    showNotification({
+                      message: "Download Completed",
+                      title: i18n.t<string>("toasts.downloadcomplete"),
+                      color: "teal",
+                    });
+                    deleteDownloadFile(fileUrl)
```

Comment:
I don't think the file is downloaded at this point in time. The "download" code above is to create a link in the DOM and click it. Browser then has to complete the transfer before we can remove the file from the backend.

Reply:
hey it may look like this but it actually does not work like the download api function gets the download file and stores it in the browsers local storage and when we click the the button it just sends it to the device . I am not sure but i think my code should work like that as i am calling the  link.remove(); in the .then() block after the `getDownloadFile` so the response is already stored is what i feel .
@derneuere  what do you think ?

Reply:
@AnkurPrabhu is right, it currently works. But I also do not understand how :sweat_smile: Right now, the file gets deleted from the file system as soon as the download starts, but it somehow can finish up correctly even with a large download, which takes a minute on localhost. Maybe nginx saves it into it's memory before providing the actual file, but I am not sure how that works.

I think we can merge it and if we figure out how to do it differently, we can then implemented in another PR.